### ISSUE TITLE
language_server/host#save expect version but it's not in the spec

### ISSUE THREAD

**Issue by hlidotbe** [castwide/solargraph#73]
Hi,

I'm trying to make diagnostics between Solargraph and LanguageClient-neovim work but there's an error when saving documents.

On save the following error is generated:
```
comparison of Integer with nil failed
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/library.rb:99:in `>'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/library.rb:99:in `overwrite'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/language_server/host.rb:164:in `block in save'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/language_server/host.rb:155:in `synchronize'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/language_server/host.rb:155:in `save'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/language_server/message/text_document/did_save.rb:7:in `process'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/language_server/host.rb:77:in `start'
/usr/local/lib/ruby/gems/2.5.0/gems/solargraph-0.23.5/lib/solargraph/language_server/transport/stdio.rb:48:in `block in process'
```

Looking into it I see it expect a version parameter which is not sent. LanguageClient-neovim only sends the filename (https://github.com/autozimu/LanguageClient-neovim/blob/cd34dd98c27880adfd17f31937a2f167a6f2c1a6/autoload/LanguageClient.vim#L594) which seems fine by the specs (https://microsoft.github.io/language-server-protocol/specification#textDocument_didSave). This, in turn, prevent the change queue from firing and so (I assume) the diagnostics from running.

Completion works mostly fine from my limited testing.

**Comment by castwide** [comment]
It looks to me like the server might not need to do anything at all in response to the textDocument/didSave notification. The fact that the current implementation works with the VS Code language client is coincidental. We should be able to eliminate the Host#save call from the DidSave message.

**Comment by hlidotbe** [comment]
That's more or less what I inferred. Disabling it when version is not provided doesn't seems to alter functionalities as most if not all is handled by the DidChange message (which is correctly versioned)

**Comment by castwide** [comment]
Cool. I've been testing it without the Host#save call and haven't seen any problems. The change is in the master branch.

**Comment by castwide** [comment]
The changes to Host#save are published in gem version 0.23.6.
